js/exporter/jspdf/v3/rows_splitting.js

```diff
@@ -1,37 +1,41 @@
 import { isDefined } from '../../../core/utils/type';
 import { extend } from '../../../core/utils/extend';
 
-function splitRowsInfosHorizontally(rowsInfosByPages, splitByColumns) {
-    if(!isDefined(splitByColumns)) {
-        return rowsInfosByPages;
+function applySplitting(pdfCellsInfo, options) {
+    if(!isDefined(options.horizontalSplitWidth)) {
+        return [ pdfCellsInfo ];
     }
 
-    const newRowsInfosByPages = [];
-    const columnIndexes = splitByColumns;
-
-    for(let i = 0; i <= columnIndexes.length; i++) {
-        const newRowsInfos = [];
-
-        rowsInfosByPages.forEach((rowsInfos) => rowsInfos.forEach(rowInfo => {
-            const newRowsInfo = extend({}, rowInfo);
-            newRowsInfo.cells = [];
-
-            const startColumn = i === 0 ? 0 : columnIndexes[i - 1];
-            const length = i < columnIndexes.length ? columnIndexes[i] - startColumn : rowInfo.cells.length - startColumn;
-            if(length > 0) {
-                for(let cellIndex = startColumn; cellIndex < startColumn + length; cellIndex++) {
-                    newRowsInfo.cells.push(rowInfo.cells[cellIndex]);
-                }
-                newRowsInfos.push(newRowsInfo);
+    const pdfCellsInfoByPage = [
+        [] // Empty Page
+    ];
+
+    pdfCellsInfo
+        .forEach(pdfCellInfo => {
+            const { _rect, pdfRowInfo, gridCell, ...pdfCell } = pdfCellInfo;
+            const _newRect = extend({}, pdfCellInfo._rect);
+            const splitWidth = options.horizontalSplitWidth;
+
+            let pageIndex = 0;
+            if((_rect.x + _rect.w) > splitWidth) {
+                const moveToPage = Math.floor((_rect.x + _rect.w) / splitWidth);
+                pageIndex += moveToPage;
```

Comment:
'pageIndex += Math.floor((_rect.x + _rect.w) / splitWidth);' ???

- pageWidth: 11
- cells: [{x:0, w:10}, {x:10, w: 10}, {x:20, w: 10] }
- pageIndex should be [0, 1, 2] but with 'pageIndex +=Math.floor((_rect.x + _rect.w) / splitWidth);' it will be [0+ (0+10)/11 = 0, 0 + (10+10)/11 = 1, 1 + (20+10)/11 = 3]